internal/pkg/mount/system.go

```diff
@@ -195,21 +199,32 @@ func SystemPartitionMount(r runtime.Runtime, label string, opts ...Option) (err
 		return err
 	}
 
+	mountpointsMutex.Lock()
+	defer mountpointsMutex.Unlock()
+
 	mountpoints[label] = mountpoint
 
 	return nil
 }
 
 // SystemPartitionUnmount unmounts a system partition by the label.
 func SystemPartitionUnmount(r runtime.Runtime, label string) (err error) {
-	if mountpoint, ok := mountpoints[label]; ok {
-		err = mountpoint.Unmount()
-		if err != nil {
-			return err
-		}
+	mountpointsMutex.RLock()
```

Comment:
I feel `RWMutex` is an overkill for the sake of a single ready here, but it doesn't matter much either way